### PR TITLE
Add option to use hard coded time dep maps in Sphere

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

Adds the ability to choose to use a TimeDependence or hard-coded time dependent maps in the Sphere domain creator. To have no time dependent maps, still specify `None`. Currently, the hard coded maps are

- Inner-most shell: a size and shape map from grid to distorted frame, identity map from distorted to inertial frame
- All other shells: no distorted frame. Only an identity map from grid to inertial
- Inner cube: Cannot have an inner cube (at the moment) when using hard coded time dependent maps

In the future we can add more maps and change where we place them, but this is sufficient for me to test shape control on a single BH while maintaining the flexibility of having TimeDependences for others.

### Upgrade instructions

If you are using the `Sphere` domain creator, the name of the time dependent options has changed from
```yaml
Sphere:
  ...
  TimeDependence:
    ...
```
to
```yaml
Sphere:
  ...
  TimeDependentMaps:
    ...
```

To specify no time dependent maps, still use `None` as before like
```yaml
Sphere:
  ...
  TimeDependentMaps: None
```

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Domain/Creators/SphereTimeDependentMaps.hpp
Comment (wthrowe) [inline src/Domain/Creators/SphereTimeDependentMaps.hpp]: What is the reason for composing with an identity map?  Please add a comment explaining.

File: src/Domain/Creators/SphereTimeDependentMaps.cpp
Comment (wthrowe) [inline src/Domain/Creators/SphereTimeDependentMaps.cpp]: Const these.  The third argument is already done.

File: src/Domain/Creators/Sphere.cpp
Comment (wthrowe) [inline src/Domain/Creators/Sphere.cpp:7]: Typo in the commit message: "Mpas".

File: src/Domain/Creators/Sphere.cpp
Comment (wthrowe) [inline src/Domain/Creators/Sphere.cpp]: Remove?

File: src/Domain/Creators/Sphere.cpp
Comment (wthrowe) [inline src/Domain/Creators/Sphere.cpp]: [optional] If you're just going to branch on the type anyway, it's I think it's cleaner to skip the visit and just do the access directly:
```c++
if (std::holds_alternative<sphere::TimeDependentMapOptions>(
        time_dependent_options_.value())) {
  return std::get<sphere::TimeDependentMapOptions>(
      time_dependent_options_.value())
      .create_functions_of_time(initial_expiration_times);
} else {
  return std::get<
      std::unique_ptr<domain::creators::time_dependence::TimeDependence<3>>>(
          time_dependent_options_.value())
          ->functions_of_time(initial_expiration_times);
}
```
(Type aliases and local reference variables might make this more readable.)

File: src/Domain/Creators/Sphere.hpp
Comment (wthrowe) [inline src/Domain/Creators/Sphere.hpp:153]: Add a comment somewhere in `TimeDependentMapOptions` reminding people to update the documentation here when they change that class.

File: tests/Unit/Domain/Creators/Test_Sphere.cpp
Comment (wthrowe) [inline tests/Unit/Domain/Creators/Test_Sphere.cpp]: Omit the `Context` argument.

File: tests/Unit/Domain/Creators/Test_Sphere.cpp
Comment (wthrowe) [inline tests/Unit/Domain/Creators/Test_Sphere.cpp]: [optional] The second `time_dependent` is redundant.  If it were false we would never get there.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: src/Domain/Creators/SphereTimeDependentMaps.hpp
Comment (knelli2) [reply src/Domain/Creators/SphereTimeDependentMaps.hpp]: I was under the assumption that since I (possibly) have a distorted -> inertial map which is just the identity, that I also need to include it in the overall grid -> inertial map so that they match. Is this not the case?

Comment (nilsvu) [reply src/Domain/Creators/SphereTimeDependentMaps.hpp]: you do not!

**Comment by knelli2** [comment]
Squashed in

**Review by nilsvu — DISMISSED** [review]

**Comment by knelli2** [comment]
We should hold off on merging this PR until #4920 is merged because that's what I'll be using.

**Review by wthrowe — DISMISSED** [review]
Marking dependent on #4920.

**Comment by knelli2** [comment]
Rebased now that #4920 and #4931 are merged.

File: tests/Unit/Domain/Creators/Test_Sphere.cpp
Comment (nilsvu) [inline tests/Unit/Domain/Creators/Test_Sphere.cpp]: delete seed

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: tests/Unit/Domain/Creators/Test_Sphere.cpp
Comment (knelli2) [reply tests/Unit/Domain/Creators/Test_Sphere.cpp]: Oops!

**Review by wthrowe — APPROVED** [review]

**Review by nilsvu — APPROVED** [review]

**Pull request merged**
